### PR TITLE
Edu.gordillo/exercise three

### PR DESCRIPTION
**Pull request by fluty84**
Done

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,35 +1,93 @@
...
-### Business Decisions
+(If you need to install Doker go to => [link](https://www.docker.com/get-started/) )
 
-In addition to testing user's data sent, I listen messsageapp error response to translate it to a more "user-friendly" language with error details, In case you have made a successful call, but the messageapp does not work properly internally. 
+```bash
+  docker compose up
+```
 
+### Docker-Compose structure
+
```
Comment (joskar) [inline README.md]: Why do you need this in the readme ?
As it is already in the  docker-compose file

File: models/message.model.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            
+        },
+        number: {
+            type: Number,
+           
+        },
+        
+        status: {
+            type: String, 
+            enum: ["Message Sent", "Message not sent", "Message sent but not confirmed"],
```
Comment (joskar) [inline models/message.model.js:20]: Nice

File: .gitignore [hunk 3]
```diff
@@ -1,4 +1,6 @@
 node_modules
 /node_modules
 */node_modules
-.DS_Store
\ No newline at end of file
+.DS_Store
+/data
+data
```
Comment (joskar) [inline .gitignore:6]: nice

File: README.md (hunk 1 above)
Comment (fluty84) [reply README.md]: Ok, I delete-it!

File: models/message.model.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+            default: "Message not sent"
```
Comment (mariaalons) [inline models/message.model.js:21]: Why choose the default as `Message not sent` ?

File: models/message.model.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        } 
+    },
+    {
+        timestamps: true,
+    }
+);
+
+const Message = model("Message", messageSchema)
+
+Message.syncIndexes()
```
Comment (mariaalons) [inline models/message.model.js:31]: What  `Message.syncIndexes()` does?

File: routes/messages.routes.js [hunk 6]
```diff
@@ -1,57 +1,92 @@
...
+    const { destination, message, number } = req.body
 
     const body = message
 
     if (!destination && !body) {
         return res.status(400).json({ message: "Need destination & message keys" })
     }
-    if (destination && typeof destination !== "string" || body && typeof body !== "string"){
-        return res.status(400).json({message: "values only can be strings"})
+    if (destination && typeof destination !== "string" || body && typeof body !== "string") {
```
Comment (mariaalons) [inline routes/messages.routes.js:24]: How does this condition work? is `"string"` a type?

**Review by mariaalons — COMMENTED** [review]
Good solution 💪

File: models/message.model.js (hunk 5 above)
Comment (fluty84) [reply models/message.model.js:31]: If I change at anytime of the project development the model, In the next call to the DB, all documents all will update with the new properties

File: models/message.model.js (hunk 4 above)
Comment (fluty84) [reply models/message.model.js:21]: Mmm maybe 🤔 you are right and I should suppose that "Message Sent" will be the most common action, so must to be default one...

File: routes/messages.routes.js (hunk 6 above)
Comment (fluty84) [reply routes/messages.routes.js:24]: Yes! Using typeof resolves "string" when you pass a String :

https://developer.mozilla.org/es/docs/Web/JavaScript/Reference/Operators/typeof

**Pull request merged**